### ISSUE TITLE
Detect missing lenient scopes configuration

### ISSUE THREAD

**Issue by alexolivier** [cerbos/cerbos#2330]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

When compiling policies, if there is a nested scope but the [lenient scopes](https://docs.cerbos.dev/cerbos/latest/configuration/engine.html#lenient_scopes) flag isn't set in the server config or the test suite show a message to the user suggesting this is why things are working as expected

**Comment by naaa760** [comment]
Hello @alexolivier :)

`
Hello please include a check to see if the "lenient scopes" flag is absent when using nested scopes during policy compilation. Show the user a clear warning message if the flag is not set and advise them to enable it in the server configuration or test suite. This will guarantee correct handling of nested scopes in policies and avoid confusion.
`

**Comment by haines** [comment]
I don't understand this one. As it says in [the docs](https://docs.cerbos.dev/cerbos/latest/configuration/engine.html#lenient_scopes)

> This setting only affects how Cerbos treats missing leaf scopes when searching for policies.

How could we know at compile time if the user is going to query at run time for a leaf scope that doesn't exist?

The only thing we can check at compile time is if there are any broken scope chains. We already do that (and it's independent of whether lenient scopes are enabled).

**Comment by alexolivier** [comment]
Agreed for compile time - where this got run into is specifically around tests which don't have the flag enabled. We have the fixtures and so know the scopes.

**Comment by alexolivier** [comment]
Maybe something even simplier. "You're policies contain scopes and the test suite is running with leninetScopesSearch ENABLED/DISABLED"

**Comment by haines** [comment]
I would be reluctant to print a message if everything is fine, except maybe in verbose mode.
